### PR TITLE
Update circleCI scripts and image

### PR DESCRIPTION
**Pull request by vissarion**
Following the discussion [here](https://github.com/boostorg/geometry/discussions/1155) this PR proposes:

- to update the circleCI image to a next-gen convenience image (read [this](https://circleci.com/docs/next-gen-migration-guide/) for the advantages of this change). Moreover, following the discussion [here](https://github.com/boostorg/geometry/pull/1147) the compiler is set to gcc-6 to possible catch library issues,
- add ~~a closing job to circleCI to cover~~ missing tests.

### CODE REVIEW

**Review by barendgehrels — APPROVED** [review]
I'm not an expert here.
I welcome improvements so I approve this.

File: .circleci/config.yml [hunk 1]
```diff
@@ -416,6 +428,15 @@ jobs:
           root: ~/project
           paths:
             - shared-coverage/index_detail
+  remaining_tests:
+    <<: *config
+    steps:
+      - *attach_workspace
+      - run: ./$BOOST_DIR/libs/geometry/.circleci/run_test.sh
```
Comment (awulkiew) [inline .circleci/config.yml:435]: What was the intention here? This script requires parameters, without it it will not run anything.

**Comment by awulkiew** [comment]
Thanks for this update!

Btw, I'd be ok with using the default gcc version. It's probably more commonly used (I assume it's ubuntu 20.04) and requires to do less manual steps which have tendency to fail.

FYI, currently the coverage step is failing due to some issues with package repositories so this change could fix it, maybe.

Have you tested whether or not it works with your fork by pushing develop or master branch and checking at CircleCI?
Even if you havent I'd be ok with pushing this change and finding out this way. In the worst case we could always reverse it.

**Comment by vissarion** [comment]
Hi @awulkiew thanks for the review. I revised my PR by adding a more straightforward solution, i.e. manually adding the missing tests. The previous approach by adding a closing job was end up too complicated (since I had to add also special cases to the `run_test.sh` and I think it doesn't worth the trouble. Now I am running circleci tests in my fork.

**Comment by vissarion** [comment]
There is also this related PR https://github.com/boostorg/geometry/pull/1154 where I need your feedback.

**Comment by awulkiew** [comment]
Thanks, I'm ok with merging it.

Regarding testing in your fork, AFAIR you have to push them to your develop branch because otherwise minimal test is tested. You can also modify the script temporarily to always run it for all branches. Unless I'm mistaken.

**Review by awulkiew — APPROVED** [review]

**Pull request merged**